delphi-frontend/src/main/java/au/com/integradev/delphi/msbuild/DelphiProjectHelper.java

```diff
@@ -366,7 +367,11 @@ public InputFile getFile(String path) {
   }
 
   public InputFile getFileFromBasename(String basename) {
-    return fs.inputFile(fs.predicates().hasFilename(basename));
+    return fs.inputFile(
+        fs.predicates()
+            .or(
+                fs.predicates().hasFilename(basename),
```

Comment:
The built-in `hasFileName` predicate is better-optimized than our new one, which is why I wanted to use a combination of the two.
This might be premature optimization, I'll test the performance difference in a real-world scan.

Reply:
It was very much *not* a premature optimization - so much so that I'm going to take an entirely different approach.